cpp/src/arrow/status.h

```diff
@@ -101,13 +102,25 @@ enum class StatusCode : char {
   CodeGenError = 40,
   ExpressionValidationError = 41,
   ExecutionError = 42
+
+};
+
+enum class Subsystem : char {
```

Comment:
Does Flight also warrant a subsystem? It would be nice to express errors like unauthorized, service unavailable/overloaded, etc. (tagging my work account @lihalite)

Reply:
Going to remove this abd go with the approach suggested by Antoine

Reply:
@lihalite when this lands, I think some codes should probably be top level (unavailable seems generic across multiple service type things) but some might be flight specific.

Reply:
Ah ok, great. I started sketching out a set of codes [for Flight/Java](https://github.com/lihalite/arrow/pull/7/files#diff-dbd9be8475cffe0a29ae8ce317320cc0), I can pick up the C++/Python side too once this lands.